app/controllers/concerns/foreman/controller/discovered_extensions.rb

```diff
@@ -54,6 +54,8 @@ def perform_auto_provision original_host, rule
     host.name = host.render_template(rule.hostname) unless rule.hostname.empty?
     # fallback to the original if template did not expand
     host.name = original_name if host.name.empty?
+    # explicitly set all inheritable attributes from hostgroup
+    host.attributes = host.apply_inherited_attributes(hostgroup_id: rule.hostgroup_id)
```

Comment:
Please call the older `set_hostgroup_defaults` too. It will set domain, os, architecture, medium, ptable and subnet.